test/integration/socketio.test.js

```diff
@@ -14,65 +14,155 @@ describe('Socket.io authentication', function() {
 
   before(function(done) {
     this.server = app.listen(PORT);
-    this.server.once('listening', () => {
-      socket = io(`http://localhost:${PORT}`);
-      done();
-    });
+    this.server.once('listening', () => done());
+  });
+
+  beforeEach(function() {
+    socket = io(`http://localhost:${PORT}`);
   });
 
   after(function() {
     this.server.close();
   });
 
   it('returns not authenticated error for protected endpoint', done => {
+    socket.emit('todos::get', 'laundry', e => {
+      delete e.stack;
+      delete e.type;
+
+      expect(e).to.deep.equal({
```

Comment:
Minor comment. I usually just check for the error code. That way if we ever change any of that other stuff we don't need to update a ton of tests.